wordpress-mobile/WordPress-Android
# Part-3: Compiler Warnings as Errors - WordPress Module - FragmentManager

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/17253 partially. (Sub-ticket of https://github.com/wordpress-mobile/WordPress-Android/pull/17246)

To test:
- [x] [RoleSelectDialogFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/people/RoleSelectDialogFragment.java):
  1. Have at least one site.
  2. Navigate to "My Site" tab.
  3. Click on "More".
  4. Click on "People".
  5. On top-left corner, click on the "+" button.
  6. You should have a form in-front of you that is used to invite people.
  7. Click on the "Role" field.
  8. A dialog should pop us as expected.
- [x] [EditPostSettingsFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java):
   1) Have a site with a post.
   2) Navigate to "My Site" and click on "Posts" to see all your posts. Open any post in edit mode.
   3) Click on dropdown button on top-left.
   4) Click "Post settings".
   5) Test this screen out and click on various options to open dialogs or activities. Do not go further (Testing individual activity or dialog) as the testing those parts is redundant.
- [x] [MyProfileFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/prefs/MyProfileFragment.java):
  1.  Go to "Me" tab using bottom navigation.
  2. Click on "My Profile".
  3. Check every field with internet on and off to ensure toast message is displayed as intended.
- [x] [SiteSettingsTagListActivity.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsTagListActivity.java) + [SiteSettingsTagDetailFragment](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsTagDetailFragment.java):
   1. Navigate to "My Site".
   2. Click on "More".
   3. Click on the option with "Site Settings".
   4. Find a setting called "Tags" and open it.
   5. Create a tag with any title and description if you don't have any.
   6. Click on a tag, navigate back, click on it again and delete the tag to verify the fragment's behaviour.
- [ ] [SiteSettingsFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/prefs/SiteSettingsFragment.java):

   No changes here as migrating to AndroidX Preference is the only way to migrate from getFragmentManager. [This issue](https://github.com/wordpress-mobile/WordPress-Android/issues/17962) dictates the progress of the problem I just mentioned. So, for now, this file will remain as is.
- [x] [ImageSettingsDialogFragment.java](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/libs/editor/src/main/java/org/wordpress/android/editor/ImageSettingsDialogFragment.java):

   1.   This dialog isn't being used anywhere as of now. So, to test it, enter the following code below in `MySiteFragment's onCreate`.
   ```
   ImageSettingsDialogFragment imageSettingsDialogFragment = new ImageSettingsDialogFragment();
   imageSettingsDialogFragment.show(parentFragmentManager, 
   ImageSettingsDialogFragment.IMAGE_SETTINGS_DIALOG_TAG);
   ```
    7.  A Dialog should pop up as expected.


## Regression Notes
1. Potential unintended areas of impact
    None

8. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual Testing

9. What automated tests I added (or what prevented me from doing so)
   None

PR submission checklist:


UI Changes testing checklist:
